src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentMarkupBlockPass.cs

```diff
@@ -327,12 +327,17 @@ public override void VisitHtmlAttribute(HtmlAttributeIntermediateNode node)
                     return;
                 }
 
-                Builder.Append("=\"");
+                // We examine the node.Prefix (e.g. " onfocus='" or " on focus=\"")
+                // to preserve the quote type that is used in the original markup.
+                var quoteType = node.Prefix.EndsWith("'", StringComparison.Ordinal) ? "'" : "\"";
```

Comment:
Could also be chars?

```suggestion
                var quoteType = node.Prefix.EndsWith('\'') ? '\'' : '"';
```

Reply:
Oh looks like https://github.com/dotnet/runtime/blob/b90631f64809a77beb7a6aef787f0254956d522b/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L603-L607 is not available?!

Reply:
Yeah -- good reco. This targets netstandard2.0 and the overload isn't available there. Ah well...